ch04packaging/03Packaging.ipynb.py

```diff
@@ -82,22 +82,38 @@
 # %autoreload 2
 
 # %% [markdown]
-# ### Using setuptools
+# ### Using pyproject.toml
 
 # %% [markdown]
+# Since June 2020, python's recommendation for creating a package is to specify package information in a `pyproject.toml` file.
+# Older projects used a `setup.py` file instead - and in fact the new `pyproject.toml` file in many ways mirrors this old format.
+# A lot of projects and packages have not yet switched over from `setup.py` to `pyproject.toml`, so don't be surprised to see a mixture of the two formats when you're looking at other people's packages.
+
+# %% [markdown]
+# For our `greetings` package, right now we are adding only the name of the package and its version number.
+# This information is included in the `project` section of our `pyproject.toml` file.
+#
+# But we also need to tell users how to build the package from these specifications.
+# This information is specified in the `build-system` section of our `toml` file.
+# In this case, we'll be using `setuptools` to build our package, so we list it in the `requires` field.
+# We also need `setuptools_scm[toml]` so that `setuptools` can understand the settings we give it in our `.toml` file, and `wheel` to make the package distribution.
 #
-# To make python code into a package, we need to write a `setup.py` file. For now we are adding only the name of the package and its version number.
+# Finally, we can set specific options for `setuptools` using additional sections in `pyproject.toml`: in this case, we will tell `setuptools` that it needs to find **and include** all of the files in our `greetings` folder.
 
 # %%
-# %%writefile greetings_repo/setup.py
+# %%writefile greetings_repo/pyproject.toml
+
+[project]
+name = "Greetings"
+version = "0.1.0"
+
+[build-system]
+requires = ["setuptools", "setuptools_scm[toml]>=6.2", "wheel"]
 
-from setuptools import setup, find_packages
+[tool.setuptools.packages.find]
+include = ["greetings*"]
 
-setup(
-    name="Greetings",
-    version="0.1.0",
-    packages=find_packages(),
-)
+[tool.setuptools_scm]
```

Comment:
Do we need this entry?

Reply:
We do - if you depend on `setuptools_scm`, you need to specify `tools.setuptools_scm` in the `pyproject`, even if you want to use the defaults :cry: 

Though in practice, we're not using dynamic versioning here so we could just remove `setuptools_scm` from the build-requires, and also remove this line.